### PR TITLE
Implement devfile test for Java Spring Boot

### PR DESCRIPTION
**Pull request by ScrewTSW**
### What does this PR do?
* Linter fix for Terminal class - replacing `"` with `'`
* Adding devfile test for Java Spring Boot devfile

### What issues does this PR fix or reference?
#18166

### How to test this PR?
* Go into Eclipse repository che/tests/e2e
* Run `npm i` and verify that all packages install without errors
* Run `tsc` and verify that no errors are present
* Export following variables:
  * `TS_SELENIUM_BASE_URL`
  * `TS_SELENIUM_MULTIUSER=true`
  * `NODE_TLS_REJECT_UNAUTHORIZED=0`
  * `TS_SELENIUM_LOG_LEVEL="TRACE"`
  * `TS_SELENIUM_USERNAME`
  * `TS_SELENIUM_PASSWORD`
* Run `npm run test-java-vertx` to execute the smoke test devfile and verify that the tests are working correctly or run entire suite using `npm run test-all-devfiles`
* Tests should pass without a failure (you may encounter stop and delete workspace failures if not using ephemeral mode by default)

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: tests/e2e/tests/devfiles/JavaSpringBoot.spec.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        workspaceHandling.createAndOpenWorkspace(stack);
+        projectAndFileTests.waitWorkspaceReadiness(workspaceSampleName, workspaceRootFolderName);
+    });
+
+    suite('Test opening file', async () => {
+        // opening file that soon should give time for LS to initialize
+        projectAndFileTests.openFile(fileFolderPath, tabTitle);
+    });
+
+    suite('Validation of workspace build and run', async () => {
```
Comment (Katka92) [inline tests/e2e/tests/devfiles/JavaSpringBoot.spec.ts]: It says it's testing build and run, but only the build command is executed.

File: tests/e2e/tests/devfiles/JavaSpringBoot.spec.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        codeExecutionTests.runTask(taskName, 240_000);
+        codeExecutionTests.closeTerminal(taskName);
+    });
+
+    suite('Language server validation', async () => {
+        commonLsTests.autocomplete(tabTitle, 32, 56, 'args : String[]');
+        commonLsTests.errorHighlighting(tabTitle, 'error_text', 30);
+        commonLsTests.codeNavigation(tabTitle, 32, 23, codeNavigationClassName, 30_000); // extended timout to give LS enough time to start
```
Comment (Katka92) [inline tests/e2e/tests/devfiles/JavaSpringBoot.spec.ts]: The timeout is not needed there as commands are running, right?

**Review by Katka92 — COMMENTED** [review]
I have a few questions, probably no big problems.

File: tests/e2e/tests/devfiles/JavaSpringBoot.spec.ts (hunk 1 above)
Comment (ScrewTSW) [reply tests/e2e/tests/devfiles/JavaSpringBoot.spec.ts]: thanks for the heads up

File: tests/e2e/tests/devfiles/JavaSpringBoot.spec.ts (hunk 2 above)
Comment (ScrewTSW) [reply tests/e2e/tests/devfiles/JavaSpringBoot.spec.ts]: good point, I should be able to remove that

**Comment by ScrewTSW** [comment]
[crw-ci-test]

**Comment by ScrewTSW** [comment]
closed because of incorrect branch and issue

**Pull request closed without merging**